libs/vkd3d/command.c

```diff
@@ -65,6 +65,7 @@ static VkImageLayout d3d12_command_list_get_depth_stencil_resource_layout(const
 static void d3d12_command_list_decay_optimal_dsv_resource(struct d3d12_command_list *list,
         const struct d3d12_resource *resource, uint32_t plane_optimal_mask,
         struct d3d12_command_list_barrier_batch *batch);
+static void d3d12_command_list_end_batch(struct d3d12_command_list *list);
```

Comment:
The intention is to prevent unnecessary command reordering, for example if an app does the copies in the beginning of the frame, and then does some compute before it draws, we don't want to defer the copies until right before the draw. 

That said, it's also possible that doing the flush in ResourceBarrier() could be sufficient, and I agree with the point on maintainability.

Reply:
Ending the copy batch in begin_render_pass and dispatch seems fine too. Just want to make sure we don't have to sprinkle it in *everywhere*.

Reply:
Games split their rendering over many command buffers too, so I doubt we should be that worried about extreme overlap.

Reply:
Sounds good, will do so.

Reply:
The `end_batch` is now constrained to:

> 1. ResourceBarrier: This is where the transition from COPY_DEST to other
>    might happen, at which point the writes must be visible. This might
>    also transition away from COPY_SRC which invalidates the
>    precondition.
> 2. Copy operations. Copies to the same resource are implicitly ordered.
> 3. Draws and dispatches. These are not strictly necessary, but we don't
>    want too much command reordering so flushing here seems good.
> 4. Close. So that we don't throw commands into the void.